File: entregas/src/main/java/br/com/bota/job/NotificacaoEntregaJob.java
package br.com.bota.job;

import br.com.bota.queue.PedidoClient;
import br.com.bota.service.EntregaService;
import io.micronaut.scheduling.annotation.Scheduled;
import lombok.extern.slf4j.Slf4j;

import javax.inject.Inject;
import javax.inject.Singleton;

@Slf4j
@Singleton
public class NotificacaoEntregaJob {
    @Inject
    private EntregaService service;
    @Inject
    private PedidoClient pedidoClient;

    @Scheduled(fixedDelay = "30s", initialDelay = "10s")
    void notificarEntrega() {
        var entregas = service.confirmarEntrega();
        entregas.parallelStream().forEach(e -> pedidoClient.notificarPacoteEntregue(e.getPedidoId()));
    }
}

File: loja-kotlin/kentregas/src/main/kotlin/br/com/bota/job/NotificacaoEntregaJob.kt
package br.com.bota.job

import br.com.bota.queue.PedidoClient
import br.com.bota.service.EntregaService
import io.micronaut.scheduling.annotation.Scheduled
import jakarta.inject.Inject
import jakarta.inject.Singleton

@Singleton
open class NotificacaoEntregaJob @Inject constructor(
    private val entregaService: EntregaService,
    private val pedidoClient: PedidoClient
) {
    @Scheduled(fixedDelay = "30s", initialDelay = "10s")
    open fun notificarEntrega() {
        var entregas = entregaService.confirmarEntrega();
        entregas.forEach {
            it.pedidoId?.let { pedidoId -> pedidoClient.notificarPacoteEntregue(pedidoId) }
        }
    }
}
